### ISSUE TITLE
If i block an user, next all befriend request will create duplicates

### ISSUE THREAD

**Issue by masterix21** [hootlex/laravel-friendships#29]
I tried to use your package and i think that you've done a good work, but i found a bug (i think).

TEST CASE: user "$manOne" (model User) isn't friend with "$girlTwo" (model User)
`
$manOne->blockFriend($girlTwo);
$manOne->befriend($girlTwo);
$manOne->befriend($girlTwo);
$manOne->befriend($girlTwo);
`

If you see database table, you'll find 4 rows. I know is not normal a block to non-friend user, but if i'd like a "block" for further friendship requests from an user, this should be a solution.

Thanks for your work.

**Comment by hootlex** [comment]
Hmm, hypothetical a user shouldn't send a friend request to a user who has blocked, before unblocking him first. But in case it happens, the duplicate rows is definitely a bug. :)

In your use case it should be:

```php
$manOne->blockFriend($girlTwo);
$manOne->unblockFriend($girlTwo);
$manOne->befriend($girlTwo);
$manOne->befriend($girlTwo);
$manOne->befriend($girlTwo);
```

I will update the package, so when a user sends a friend request to a user who has blocked, it will automatically unblock him.

**Comment by hootlex** [comment]
Resolved in v1.0.18
